gno/r/launchpad_grc20/airdrop_grc20_test.gno

```diff
@@ -0,0 +1,352 @@
+				amountPerAddr: 100,
+				start:         0,
+				end:           100,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+		"Airdrop should start before it ends": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         100,
+				end:           50,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+	}
+
+	// Create tokens for testing
+	NewToken("TestNewAirdropMintableToken", "TestNewAirdropMintableToken", "noimage", 18, 21_000_000, 23_000_000, true, true)
+	NewToken("TestNewAirdropNotMintableToken", "TestNewAirdropNotMintableToken", "noimage", 18, 21_000_000, 23_000_000, false, true)
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			if test.expected.panic {
+				defer func() {
+					if r := recover(); r == nil {
+						t.Errorf("Expected panic, got none")
+					}
+				}()
+			}
+
+			airdropID := NewAirdrop(test.input.tokenName, test.input.root, test.input.amountPerAddr, int64(test.input.start), int64(test.input.end))
+			airdrop := mustGetAirdrop(airdropID)
+
+			if !test.expected.panic {
+				if airdrop.merkleRoot != test.expected.root {
+					t.Errorf("Expected root to be %s, got %s", test.expected.root, airdrop.merkleRoot)
+				}
+				if airdrop.amountPerAddr != test.expected.amountPerAddr {
+					t.Errorf("Expected amountPerAddr to be %d, got %d", test.expected.amountPerAddr, airdrop.amountPerAddr)
+				}
+				if airdrop.startTimestamp != int64(test.expected.start) {
+					t.Errorf("Expected startTimestamp to be %d, got %d", test.expected.start, airdrop.startTimestamp)
+				}
+				if airdrop.endTimestamp != int64(test.expected.end) {
+					t.Errorf("Expected endTimestamp to be %d, got %d", test.expected.end, airdrop.endTimestamp)
+				}
+			}
+		})
+	}
+}
+
+func TestClaimJSON(t *testing.T) {
+	type testClaimJSONInput struct {
+		airdropID uint64
+		proofs    string
+	}
+
+	type testClaimJSONExpected struct {
+		panic bool
+	}
+
+	type testClaimJSON struct {
+		input    testClaimJSONInput
+		expected testClaimJSONExpected
+	}
+
+	type testClaimJSONTestTable = map[string]testClaimJSON
+
+	leaves := []merkle.Hashable{
+		Leaf{[]byte("g126gx6p6d3da4ymef35ury6874j6kys044r7zlg")},
+		Leaf{[]byte("g1ld6uaykyugld4rnm63rcy7vju4zx23lufml3jv")},
+	}
+
+	tree := merkle.NewTree(leaves)
+	root := tree.Root()
+	t.Logf("Root: %s", root)
+	proofs, _ := tree.Proof(leaves[0])
+
+	erroneousProofs := []merkle.Node{
+		{[]byte("badproof")},
+		{[]byte("badproof")},
+	}
+
+	now := time.Now().Unix()
+	NewToken("TestClaimJSONAirDropToken", "TestClaimJSONAirDropToken", "noimage", 18, 21_000_000, 23_000_000, true, true)
+	airdropID := NewAirdrop("TestClaimJSONAirDropToken", root, 100, 0, 0)
+
+	proofsJSON := "["
+	for i, proof := range proofs {
+		proofsJSON += fmt.Sprintf("{\"hash\":\"%s\", \"pos\":\"%s\"}", proof.Hash(), strconv.Itoa(int(proof.Position())))
```

Comment:
I really dislike that we have to escape json like this. Hopefully in the future gno parses it better